client/main/dataset.js

```diff
@@ -895,6 +938,188 @@ const DataSetModel = Backbone.Model.extend({
+                        transform = { };
+                        nCreated++;
+                        this._readTransformPB(transform, transformPB);
+                        this.attributes.transforms.push(transform);
+                    }
+                    let nameChanged = (oldName !== transformPB.name);
+
+                    changed[i] = transformPB.name;
+                    changes[i] = {
+                        id: id,
+                        name: transform.name,
+                        oldName: oldName,
+                        nameChanged: nameChanged,
+                        created: created,
+                        formulaMessageChanged: transform.formulaMessage !== oldMessage
+                    };
+                }
+
+                for (let change of changes) {
+                    if (change.created)
+                        this.trigger('transformAdded', { id: change.id });
+                }
+
+                this.trigger('transformsChanged', { changed, changes });
+            }
+        }).catch((error) => {
+            console.log(error);
+            throw error;
+        });
+    },
+
+    removeTransforms(ids) {
+
+        let coms = this.attributes.coms;
+        let datasetPB = new coms.Messages.DataSetRR();
+        datasetPB.op = coms.Messages.GetSet.SET;
+        datasetPB.incSchema = true;
+        datasetPB.schema = new coms.Messages.DataSetSchema();
+
+        for (let id of ids) {
+
+            let transform = this.getTransformById(id);
+
+            let transformPB = new coms.Messages.DataSetSchema.TransformSchema();
+            transformPB.id = id;
+            transformPB.action = 2; // action: 0 - CREATE, 1 - UPDATE, 2 - REMOVE
+
+            datasetPB.schema.transforms.push(transformPB);
+        }
+
+        let request = new coms.Messages.ComsMessage();
+        request.payload = datasetPB.toArrayBuffer();
+        request.payloadType = 'DataSetRR';
+        request.instanceId = this.attributes.instanceId;
+
+        return coms.send(request).then(response => {
+            let changed = [];
+            let changes = [];
+
+            let transform = null;
+            for (let id of ids) {
+
+                for (let i = 0; i < this.attributes.transforms.length; i++) {
+                    transform = this.attributes.transforms[i];
+                    if (transform.id === id) {
+                        this.attributes.transforms.splice(i, 1);
+                        break;
+                    }
+
+                    changed[i] = transform.name;
+                    changes[i] = {
+                        id: id,
+                        name: transform.name,
+                        deleted: true,
+                    };
+                }
+            }
+
+            for (let change of changes)
+                this.trigger('transformRemoved', { id: change.id });
+
+            this.trigger('transformsChanged', { changed, changes });
+        }).catch((error) => {
```

Comment:
use .done() here:

https://github.com/kriskowal/q#the-end

Reply:
Is not necessary because these promises return.